.github/scripts/develocity_reports.py

```diff
@@ -0,0 +1,863 @@
+            PROJECT,
+            threshold_days=7,  # Last 7 days for regular tests
+            test_type="test"
+        )
+        
+        # Generate reports
+        problematic_tests = analyzer.get_problematic_quarantined_tests(
+            quarantined_results, 
+            QUARANTINE_THRESHOLD_DAYS,
+            MIN_FAILURE_RATE,
+            RECENT_FAILURE_THRESHOLD
+        )
+        
+        flaky_regressions = analyzer.get_flaky_test_regressions(
+            PROJECT,
+            regular_results,
+            recent_days=7,
+            min_flaky_rate=MIN_FLAKY_RATE
+        )
+        
+        cleared_tests = analyzer.get_cleared_tests(
+            PROJECT,
+            quarantined_results,
+            success_threshold=SUCCESS_THRESHOLD
+        )
+        
+        # Print reports
+        print(f"\nTest Analysis Report ({datetime.now(pytz.UTC).strftime('%Y-%m-%d %H:%M:%S')} UTC)")
+        print("=" * 100)
+        
+        # Print Flaky Test Regressions
+        print("\nFlaky Test Regressions")
+        print("-" * 50)
+        if not flaky_regressions:
+            print("No flaky test regressions found.")
+        else:
+            for test_name, details in flaky_regressions.items():
+                print(f"\n{test_name}")
+                print(f"Recent Flaky Rate: {details['recent_flaky_rate']:.2%}")
+                print(f"Historical Flaky Rate: {details['historical_flaky_rate']:.2%}")
+                print("\nRecent Executions:")
+                for entry in sorted(details['recent_executions'], key=lambda x: x.timestamp)[-5:]:
+                    print(f"  {entry.timestamp.strftime('%Y-%m-%d %H:%M')} - {entry.outcome}")
+        
+        # Print Cleared Tests
+        print("\nCleared Tests (Ready for Unquarantine)")
+        print("-" * 50)
+        if not cleared_tests:
+            print("No tests ready to be cleared from quarantine.")
+        else:
+            for test_name, details in cleared_tests.items():
+                print(f"\n{test_name}")
+                print(f"Success Rate: {details['success_rate']:.2%}")
+                print(f"Total Executions: {details['total_executions']}")
+                print("\nRecent Executions:")
+                for entry in sorted(details['recent_executions'], key=lambda x: x.timestamp):
+                    print(f"  {entry.timestamp.strftime('%Y-%m-%d %H:%M')} - {entry.outcome}")
+        
+        # Print Defective Tests (with detailed reporting restored)
+        print("\nHigh-Priority Quarantined Tests")
+        print("-" * 50)
+        if not problematic_tests:
+            print("No high-priority quarantined tests found.")
+        else:
+            sorted_tests = sorted(
+                problematic_tests.items(), 
+                key=lambda x: (x[1]['failure_rate'], x[1]['days_quarantined']),
+                reverse=True
+            )
+            
+            print(f"\nFound {len(sorted_tests)} high-priority quarantined test containers:")
```

Comment:
The count here should be the number of flaky test cases rather than flaky test classes (containers).

Reply:
Also, we should not use the term "container" in the report since it's kind of confusion. As far as I know, for our purposes a container is always a test class.

Reply:
Corrected this to test class.